File: python/cs50/cash.py
#!/usr/bin/env python3
from cs50 import get_float


def change(number, value):  # function to calculate each coin
    increment = 0
    while number >= value:
        number -= value  # calculate new value
        increment += 1  # add an additional coin
    return number, increment


n = -1
while n < 0:
    n = get_float("Change owed: ")  # prompt for input

n *= 100  # turn into pennies
coins = 0
for i in [25, 10, 5, 1]:
    result = change(n, i)  # run function
    n = result[0]  # new n value
    coins += result[1]  # add the coins

print(coins)

File: c/cs50/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int count(int n, int c);
int get_input(void);

int coins = 0;

int main(void)
{
    int n = get_input();
    n = count(n, 25); //iterates through the coins
    n = count(n, 10);
    n = count(n, 5);
    n = count(n, 1);
    printf("%i\n", coins);
}

int count(int n, int c)
{
    while (n >= c) //loops taking off one coin at a time until no more can be taken
    {
        n -= c;
        coins++;
    }
    return n;
}

int get_input(void)
{
    float input;
    do
    {
        input = get_float("Change owed: "); //loops input util it's greater than 0
    }
    while (input < 0);
    int n = round(input * 100); //changes to cents and removes any extraneous bits of a cent
    return n;
}